libvast/vast/system/index.hpp

```diff
@@ -174,6 +177,9 @@ struct index_state {
   /// Our current partition.
   partition_ptr active;
 
+  /// Our current partition.
+  size_t active_partition_indexers;
```

Comment:
I really dislike this extra bookkeeping state. It tracks information we can easily compute: the number of non-zero actor handles in `table_indexer::indexers_`. I think it's easy for this member to go out of sync with the table indexer and all it really achieves is saving us one `O(n)` computation in `reset_active_partition`. To me, that's not worth it. The size of `table_indexer::indexers_` is the number of columns in a schema and this number is going to be small in practice. For example, a "large" schema such as for Bro conn logs has ~20 columns.

So rather than using this extra state, I propose to add a member function such as `table_indexer::active_columns()` for computing the number of non-null entries in `indexers_` on demand.

Reply:
> I really dislike this extra bookkeeping state. It tracks information we can easily compute: the number of non-zero actor handles in table_indexer::indexers_.

`active_partition_indexers` is not equal to the number of valid handles in `table_indexer::indexers_`. `table_indexer::indexers_` is not modified when an INDEXER completes processing its inbound stream, but `active_partition_indexers` is. 
Counting all valid handles would not be enough to implement `table_indexer::active_columns()`, because it does not distinguish between active and finished INDEXERs. I don't see a way to access the streams through a `caf::actor` handle either. Maybe there is a way to inject a callback to `indexer_downstream_manager` that runs when a sink is finished, but it doesn't look like it.

Reply:
>  I don't see a way to access the streams through a caf::actor handle either. Maybe there is a way to inject a callback to indexer_downstream_manager that runs when a sink is finished, but it doesn't look like it.

Yeah, I don't see a straight-forward stateless solution either, even though I'd like to see one. Since you're away, @Neverlord, I'm going to accept @tobim's solution for now, but will keep track of this discussion so that we can come back to it if needed.